ibm/service/cis/data_source_ibm_cis_origin_auth.go

```diff
@@ -0,0 +1,187 @@
+// Copyright IBM Corp. 2017, 2022 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package cis
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	CisOriginAuthHostName    = "hostname"
+	CisOriginAuthRequestType = "request_type"
+)
+
+func DataSourceIBMCISOriginAuthPull() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataIBMCISOriginAuthRead,
+		Schema: map[string]*schema.Schema{
+			cisID: {
+				Type:        schema.TypeString,
+				Description: "CIS instance crn",
+				Required:    true,
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Description:      "Associated CIS domain",
+				Required:         true,
+				DiffSuppressFunc: suppressDomainIDDiff,
+			},
+			CisOriginAuthHostName: {
+				Type:        schema.TypeString,
+				Description: "Associated CIS host name",
+				Optional:    true,
+				Default:     "no_host",
+			},
+			CisOriginAuthRequestType: {
+				Type:        schema.TypeString,
+				Description: "Associated CIS Request Type",
+				Optional:    true,
+				Default:     "zone_level",
+			},
+			"origin_pull_settings_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "CIS origin auth settings enabled or disabled",
+			},
+			"origin_pull_certs_list": {
```

Comment:
No need to term the variable to include the type of variable by default indidcates its a list..Can you change this to "origin_pull_certs"